matdeeplearn/trainers/base_trainer.py

```diff
@@ -107,10 +110,13 @@ def from_config(cls, config):
         )
         scheduler = cls._load_scheduler(config["optim"]["scheduler"], optimizer)
         loss = cls._load_loss(config["optim"]["loss"])
-
         max_epochs = config["optim"]["max_epochs"]
+        max_checkpoint_epochs = config["optim"]["max_checkpoint_epochs"]
         identifier = config["task"].get("identifier", None)
         verbosity = config["task"].get("verbosity", None)
+        # pass in custom results home dir and load in prev checkpoint dir
+        save_dir = config["task"].get("save_dir", os.getcwd())
```

Comment:
Also actually why don't you switch this to be 
```suggestion
        save_dir = config["task"].get("save_dir", None)
```
because you'll need to duplicate this in the base __init__ because we don't want `save_dir` to be mandatory, but default behavior should be `os.getcwd()`

Reply:
Actually the intention here was to replace the original `run_dir` variable from previous versions, and `save_dir` can stay optional in the config but default to `os.getcwd()`. However, it would make more sense to have `save_dir` in that case be positional in the BaseTrainer `__init__` since it's guaranteed to have a value. Could we do this instead?